library/ssl_cache.c

```diff
@@ -321,6 +326,8 @@ void mbedtls_ssl_cache_free( mbedtls_ssl_cache_context *cache )
 #if defined(MBEDTLS_THREADING_C)
     mbedtls_mutex_free( &cache->mutex );
 #endif
+
+    mbedtls_zeroize( cache, sizeof(mbedtls_ssl_cache_context) );
```

Comment:
This line does the job, but I would prefer to simply set `cache->chain` to `NULL` here, because that it conveys the purpose behind this line much better to the reader.

Reply:
I don't think it's worth another commit, but in general I think our `xxx_free` functions zeroize the entire structure, so I'd prefer the original version of the PR. But for `mbedtls_ssl_cache_context` specifically, clearing `chain` is the important step.